tests/test_official.py

```diff
@@ -77,15 +78,26 @@ def check_method(h4):
     sig = inspect.signature(method, follow_wrapped=True)
 
     checked = []
-    for parameter in table:
-        param = sig.parameters.get(parameter[0])
-        assert param is not None, f"Parameter {parameter[0]} not found in {method.__name__}"
+    for tg_parameter in table:  # Iterates through each row in the table
+        param = sig.parameters.get(
+            tg_parameter[0]  # parameter[0] is first element (the param name)
+        )
+        assert param is not None, f"Parameter {tg_parameter[0]} not found in {method.__name__}"
 
         # TODO: Check type via docstring
         assert check_required_param(
-            parameter, param.name, sig, method.__name__
+            tg_parameter, param, method.__name__
         ), f'Param {param.name!r} of method {method.__name__!r} requirement mismatch!'
-        checked.append(parameter[0])
+
+        # Now we will check that we don't pass default values if the parameter is not required.
+        # parameter[2] can either be Required/Yes or a paragraph where the first word has that info
+        telegram_param_required = is_parameter_required_by_tg(tg_parameter[2])
+
+        if not telegram_param_required:
+            default_arg_none = check_defaults_type(param)
+            assert default_arg_none, f"Param {param.name!r} of {method.__name__!r} should be None"
```

Comment:
why does it matter if the parameter is required by TG? IISC it should suffice to check that all parameters that are being passed to TG are either required or have `(Default)None` as default value, no?